apps/wpcom-block-editor/src/wpcom/features/fix-block-invalidation-errors.js

```diff
@@ -11,6 +11,39 @@ import { createBlock } from '@wordpress/blocks';
  */
 import { isEditorReadyWithBlocks } from '../../utils';
 
+/**
+ * Checks if a given block object has content.
+ *
+ * Makes sure that we don't delete content when block auto-fix gives us empty
+ * block content. For example, if a paragraph block structure is invalid, its
+ * content attribute may be an empty string. However, depending on the error,
+ * the block could still be fixed via the code editor. This way, we keep blocks
+ * which do not auto-fix into a somewhat reasonable shape so that they can be
+ * manually fixed.
+ *
+ * Note that we return 'true' if we don't understand how to validate the block.
+ * This way, we continue auto-fixing other blocks if we can.
+ *
+ * @param {object} block The block to check for content.
+ * @returns bool True if the block has content. False otherwise.
+ */
+function blockHasContent( block ) {
+	// There is no content if there is no block.
+	if ( ! block ) {
+		return false;
+	}
+	switch ( block.name ) {
+		case 'core/paragraph':
+			return block.attributes?.content?.length > 0;
```

Comment:
The approach here seems reasonable to me @noahtallen ! I can help with manual testing tomorrow if other folks don't beat me to it 👍